hazelcast-sql/src/main/java/com/hazelcast/jet/sql/impl/PlanExecutor.java

```diff
@@ -293,6 +311,10 @@ SqlResult execute(ShowStatementPlan plan) {
                 JetServiceBackend jetServiceBackend = nodeEngine.getService(JetServiceBackend.SERVICE_NAME);
                 rows = jetServiceBackend.getJobRepository().getActiveJobNames().stream();
                 break;
+            case TYPES:
+                rows = new TypesStorage(getNodeEngine(hazelcastInstance)).getAllTypes().stream()
```

Comment:
I believe this one will not work properly.

The ```TypesStorage extends TablesStorage```, the ```TablesStorage``` controls if we use ```ReplicatedMap``` or ```IMap``` for catalog.  

If you create a new instance then ```storageMovedToNew == false``` an you are going to run:

```java
    private Map<String, Object> mergedStorage() {
        // ...
            if (!storageMovedToNew) {
                synchronized (mergingMutex) {
                    if (!storageMovedToNew) {
                        ReplicatedMap<String, Object> oldStorage = oldStorage();
                        oldStorage.forEach(newStorage::putIfAbsent);
                        oldStorage.destroy();
                        storageMovedToNew = true;
                    }
                }
        // ...    
}
```

for every run. Can you merge the code to ```TablesStorage``` and use the single created instance?